lms/models/application_instance.py

```diff
@@ -45,6 +45,9 @@ class ApplicationInstance(BASE):
         "OAuth2Token", back_populates="application_instance"
     )
 
+    #: A list of all the courses for this application instance.
+    courses = sa.orm.relationship("Course", back_populates="application_instance")
```

Comment:
This lets you do `application_instance.courses` to get a list of all the related courses. It also informs SQLAlchemy how to handle the related instances: if you add both a new `ApplicationInstance` and a new `Course` for that instance to the DB session at the same time, SQLAlchemy now knows to flush the `ApplicationInstance` to the DB before the `Course` to avoid an Integrity Error from Postgres

Reply:
It's needed for the factory to work. Otherwise SQLAlchemy will crash on saving a factory-created application instance and courses to the DB because it doesn't understand that it has to serialize the application instance first because the courses depend on it:

```python
>>> course = factories.Course(consumer_key=factories.ApplicationInstance().consumer_key)
>>> request.tm.commit()
IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "course" violates foreign key constraint "fk__course__consumer_key__application_instances"
DETAIL:  Key (consumer_key)=(Hypothesis69240600f272c337b13aa2e257bc5bbd) is not present in table "application_instances".
```